planner_cspace/src/planner_3d.cpp

```diff
@@ -1872,9 +1914,13 @@ class Planner3dNode
       poses.poses.push_back(pose);
     }
     pub_path_poses_.publish(poses);
-
+    if (!start_pose_predictor_.getPreservedPath().poses.empty())
+    {
+      pub_preserved_path_poses_.publish(start_pose_predictor_.getPreservedPath());
+    }
     const std::list<Astar::Vecf> path_interpolated =
         model_->path_interpolator_.interpolate(path_grid, 0.5, local_range_);
+    path.poses = start_pose_predictor_.getPreservedPath().poses;
     grid_metric_converter::grid2MetricPath(map_info_, path_interpolated, path);
```

Comment:
I originally didn't expected to use `grid2MetricPath` to append poses to existing path.
It would be better to rename it to something like `grid2MetricPathAppended`/`grid2MetricPathWithAppend`.

Reply:
Renamed to `appendGridPath2MetricPath`